services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java

```diff
@@ -253,25 +269,73 @@ public void runGetTables(FlightSql.CommandGetTables commandGetTables,
 
     final UserProtos.GetTablesResp getTablesResp = responseHandler.get();
 
-    try (VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA,
-      allocator)) {
+    final boolean includeSchema = commandGetTables.getIncludeSchema();
+    final Map<String, List<Field>> tableToFields = new HashMap<>();
+
+    if (includeSchema) {
+      final UserBitShared.ExternalId columnRunExternalId = ExternalIdHelper.generateExternalId();
+
+      final UserProtos.GetColumnsReq.Builder columnBuilder = UserProtos.GetColumnsReq.newBuilder();
+      final UserRequest columnsRequest = new UserRequest(UserProtos.RpcType.GET_COLUMNS, columnBuilder.build());
+
+      final CancellableUserResponseHandler<UserProtos.GetColumnsResp> columnResponseHandler =
+        new CancellableUserResponseHandler<>(runExternalId, userSession, workerProvider, isRequestCancelled,
+          UserProtos.GetColumnsResp.class);
+
+      workerProvider.get().submitWork(columnRunExternalId, userSession, columnResponseHandler,
+        columnsRequest, TerminationListenerRegistry.NOOP);
+
+      final UserProtos.GetColumnsResp getColumnsResp = columnResponseHandler.get();
+
+      final IntStream columnsRange = IntStream.range(0, getColumnsResp.getColumnsCount());
+
+      columnsRange.forEach(i -> {
+        final UserProtos.ColumnMetadata columns = getColumnsResp.getColumns(i);
+
+        final String tableName = columns.getTableName();
+        final List<Field> fields = tableToFields.computeIfAbsent(tableName, tableName_ -> new ArrayList<>());
+
+        final Field field = new Field(
```

Comment:
Shouldn't we include the nullable information?

Reply:
I think we should for consistency. If a query is returning this values so the getTables